h2/src/test/org/h2/test/db/TestCompatibilitySQLServer.java

```diff
@@ -0,0 +1,76 @@
+package org.h2.test.db;
+
+import org.h2.test.TestBase;
+import org.h2.test.TestDb;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * Test MSSQLServer compatibility mode.
+ */
+public class TestCompatibilitySQLServer extends TestDb {
+
+    /**
+     * Run just this test.
+     *
+     * @param s ignored
+     */
+    public static void main(String... s) throws Exception {
+        TestBase test = TestBase.createCaller().init();
+        test.test();
+    }
+
+    @Override
+    public void test() throws Exception {
+        testDiscardTableHints();
+        testUseIdentityAsAutoIncrementAlias();
+    }
+
+    private void testDiscardTableHints() throws SQLException {
+        deleteDb("sqlserver");
+
+        Connection conn = getConnection("sqlserver;MODE=MSSQLServer");
```

Comment:
Connection should be closed somewhere. The same problem in `testUseIdentityAsAutoIncrementAlias()`.

Also it's better to delete database not only before execution of test case, but also after it. Actually you can do it in `test()` method and use shared connection in both test cases, there is no reason to use a separate connection in them now.

Reply:
FYI, I actually used `TestCompatibilityOracle` as a reference point (and still forgot to close the connections) - it deletes the db before each test only. Might be worth checking out.